kernel/src/deferred_call.rs

```diff
@@ -217,15 +219,18 @@ impl DeferredCall {
     fn register_internal_non_generic(&self, handler: DynDefCallRef<'static>) {
         if let Some(defcalls_cell) = DEFCALLS.get() {
             defcalls_cell.map(|defcalls| {
-                if self.idx >= defcalls.len() {
-                    // This error will be caught by the scheduler at the beginning of
-                    // the kernel loop, which is much better than panicking here, before
-                    // the debug writer is setup. Also allows a single panic for
-                    // creating too many deferred calls instead of NUM_DCS panics (this
-                    // function is monomorphized).
-                    return;
+                if let Some(idx_plus_one) = self.idx_plus_one {
+                    let idx = usize::from(idx_plus_one) - 1;
+                    if idx >= defcalls.len() {
+                        // This error will be caught by the scheduler at the beginning of
+                        // the kernel loop, which is much better than panicking here, before
+                        // the debug writer is setup. Also allows a single panic for
+                        // creating too many deferred calls instead of NUM_DCS panics (this
+                        // function is monomorphized).
+                        return;
+                    }
+                    defcalls[idx].set(handler);
```

Comment:
This can be:

```rust
let Some(c) = defcalls.get(idx) else {
    // This error will be caught by the scheduler ...
    return;
};
c.set(handler);
```
(might need that `Some(c)` to be `Some(ref mut c)` or similar). This pattern avoids a panic path.

Reply:
This is actually a comment for the other PR.